tcms/rpc/tests/test_testplan.py

```diff
@@ -226,3 +227,116 @@ def test_ignores_existing_mappings(self):
                         case=case_id
                     ).count()
                 )
+
+
+class TestCreate(APITestCase):
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.product = ProductFactory()
+        self.version = VersionFactory(product=self.product)
+        self.plan_type = PlanTypeFactory(name='api testing')
+        self.params = {
+            'product': self.product.pk,
+            'product_version': self.version.pk,
+            'name': 'Testplan foobar',
+            'type': self.plan_type.pk,
+            'text': 'Testing TCMS',
+            'parent': None,
+        }
+
+    def test_create_plan(self):
+        plan = self.rpc_client.TestPlan.create(self.params)
+        self.assertIsNotNone(plan)
+        self.assertEqual(self.params['product'], plan['product_id'])
+        self.assertEqual(self.params['product_version'], plan['product_version_id'])
+        self.assertEqual(self.params['name'], plan['name'])
+        self.assertEqual(self.params['type'], plan['type_id'])
+        self.assertEqual(self.params['text'], plan['text'])
+        self.assertEqual(self.params['parent'], plan['parent'])
+
+    def test_create_plan_with_empty_required_field(self):
+        self.params['name'] = ''
+        with self.assertRaisesRegex(XmlRPCFault, 'This field is required'):
+            self.rpc_client.TestPlan.create(self.params)
+
+    def test_create_plan_without_permissions(self):
+        unauthorized_user = UserFactory()
+        unauthorized_user.set_password('api-testing')
+        unauthorized_user.save()
+
+        unauthorized_user.user_permissions.add(*Permission.objects.all())
+        remove_perm_from_user(unauthorized_user, 'testplans.add_testplan')
+
+        rpc_client = xmlrpc.TCMSXmlrpc(unauthorized_user.username,
+                                       'api-testing',
+                                       '%s/xml-rpc/' % self.live_server_url).server
+
+        with self.assertRaisesRegex(ProtocolError, '403 Forbidden'):
+            rpc_client.TestPlan.create(self.params)
+
+
+class TestListAttachments(APITestCase):
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.product = ProductFactory()
+        self.version = VersionFactory(product=self.product)
+        self.plan = TestPlanFactory(product_version=self.version,
+                                    product=self.product,
+                                    author=UserFactory(),
+                                    type=PlanTypeFactory(name='manual smoking'))
+
+    def test_list_attachments(self):
+        attachments = self.rpc_client.TestPlan.list_attachments(self.plan.pk)
+        self.assertIsNotNone(attachments)
```

Comment:
This is wrong:

- `self.plan` has not received any attachments uploaded before this scenario is executed so the `attachments` variable will be an empty list `[]` which is different than `None` hence the assert passes

You should attach files first and then assert that all of the attached files are on the list.